server/resp/src/main/java/org/infinispan/server/resp/RespServer.java

```diff
@@ -40,7 +41,16 @@ protected void startInternal() {
       if (!cacheManager.getCacheManagerConfiguration().features().isAvailable(RESP_SERVER_FEATURE)) {
          throw CONFIG.featureDisabled(RESP_SERVER_FEATURE);
       }
-      if (cacheManager.getCacheConfiguration(configuration.defaultCacheName()) == null) {
+      String defaultRespCacheName = configuration.defaultCacheName();
+      Configuration respCacheConfig = cacheManager.getCacheConfiguration(defaultRespCacheName);
+      if (respCacheConfig != null) {
+         ExpirationConfiguration expConfig = respCacheConfig.expiration();
+         if (expConfig.lifespan() >= 0 || expConfig.maxIdle() >= 0)
+            throw log.invalidExpiration(configuration.defaultCacheName());
+         super.startInternal();
+
+      } else {
+         super.registerAdminOperationsHandler();
```

Comment:
I would personally put this with the `if (configuration.startTransport())` line below and put a comment as to why we aren't invoking `startInternal` there. I feel this will be a bit brittle if anything changes in `startInternal`.

Reply:
I did the change, let me know @wburns if this is what you asked for

Reply:
This is what I was asking for, yes. However, you didn't add a comment.

Thinking about this further though, I think it would be a ton easier to change this slightly now that I am looking at this.

I think the startInternal method should just contain
```java
   protected void startInternal() {
      if (!cacheManager.getCacheManagerConfiguration().features().isAvailable(RESP_SERVER_FEATURE)) {
         throw CONFIG.featureDisabled(RESP_SERVER_FEATURE);
      }
      super.startInternal();
```
And then we put all the rest of the code in `startCaches`.

WDYT?

Reply:
done